src/NServiceBus.Testing/TestableMessageHandlerContext.cs

```diff
@@ -15,15 +16,15 @@ public TestableMessageHandlerContext(IMessageCreator messageCreator)
             this.messageCreator = messageCreator;
         }
 
-        public IList<TimeoutMessage<object>> TimeoutMessages { get; } = new List<TimeoutMessage<object>>();
+        public IList<SentMessage<object>> SentMessages => sentMessages.ToArray();
```

Comment:
Why `ToArray`?

Does this have to return `IList`? What is the benefit instead of just exposing an array directly or enumerable?

Btw. if you enumerate a ConcurrentQueue it will make a copy.

http://referencesource.microsoft.com/#mscorlib/system/Collections/Concurrent/ConcurrentQueue.cs,465